[Michael] iP

Dr-Octavius opened the pull request:
# Duke

> “Your mind is for having ideas, not holding them.” – David Allen (source)


Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to lear
- FAST SUPER FAST to use
All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Dr-Octavius committed:
Greet, Echo, Exit

Dr-Octavius committed:
Added comments

Dr-Octavius committed:
Add, List w/ comments

Dr-Octavius committed:
Mark as Done, Added Task Class, w/ comments

Dr-Octavius committed:
Removed Scanner Object as arguments from constructor parameters

Dr-Octavius committed:
ToDo, Event, Deadline w/ comments

Dr-Octavius committed:
Added Testing Input and expected Output on text-ui-test folder

Dr-Octavius committed:
Automated Text UI Testing

Dr-Octavius committed:
Handle Errors - Minimal

Dr-Octavius committed:
Delete

Dr-Octavius committed:
Use Enums

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+            + "What can I do for you?\n")
+            .concat(SEP_LINE);
+    // Level-6 -> A-Collections: Text Storage
+    private final List<Task> storage = new ArrayList<>();
```
Perhaps a more intuitive variable name here like `taskList`?

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+
+    // Level-1: echos user message back to user
+    // Level-5: returns the error message for unknown inputs
+    public String echo() {
```
Since the method no longer `echo`s the user input, I believe it would be more fitting to rename it.

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+        return " Bye. Hope to see you again soon!";
+    }
+
+
```
There's a redundant line here. 😅

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+        Task nextTask = new Task(text);
+        this.storage.add(nextTask);
+        return "added: ".concat(text);
+    }
```
I imagine the blank line above should be here 😂

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+    }
+
+    // Level-3: Mark items as done
+    public String done(int index) {
```
As function names are usually verbs, perhaps name the method `markAsDone` here?

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+    }
+
+    // Level-4: setting of task urgency to "to-do"
+    public String todo(String description) {
```
Similarly should this be called e.g. `addTodoTask`? I notice the same issue in the next 2 functions too.

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+
+    // Level-4: setting of task urgency to "to-do"
+    public String todo(String description) {
+        Task task = new Todo(description,state.T);
```
Having white spaces around the comma can help with readability!

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+    public int decoder(String userInput) {
+        int res;
+        switch (userInput) {
+            case "bye":
```
For `switch-case` statements the indentation should be the same for `switch` and `case` as stated by the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html)!

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+    }
+
+    // Level 3 -> A-Classes: Task class that represents all Tasks
+    private static class Task {
```
Any reason you put `Task` as an inner class? Should it be in a separate file for more clarity, and later to be put in packages too?

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+    public String list() {
+        String out = "Here are the tasks in your list:\n";
+        for (int i = 0; i < this.storage.size(); i++) {
+            if (i != 0) out = out.concat("\n");
```
This part can add `{}` brackets for the `if` statement according to the module's coding guideline.

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+
+        // overloaded constructors for separate task types
+        public Task(String description, String var, state STATE) {
+
```
I think you can remove the blank line here!

uyencfi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,325 @@
+    }
+
+    // A-Enums: Set Enums for Task type
+    protected enum state {
```
Should enum names be in PascalCase, so `State`?

uyencfi reviewed (commented):
The code can benefit from some refactoring and some tidying up to comply with the coding standard. Other than that, LGTM!

Dr-Octavius committed:
Implement Save Method

Dr-Octavius committed:
Implement Date and Time Parsing with LocalDate

Dr-Octavius committed:
Change Contructor Parameter Names

ananya-ravi-3021 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,239 @@
+import java.util.ArrayList;
+
+// A-Enums: Set Enums for Task type
+enum TASK_TYPE {
```
Perhaps name this enum using PascalCase.

```suggestion
enum TaskType {
```

ananya-ravi-3021 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,239 @@
 public class Duke {
+
+    // shows if the Duke chat bot has been activated
+    private boolean activated;
```
Should this be named to sound more like a boolean?

```suggestion
    private boolean hasActivated;
```

ananya-ravi-3021 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,239 @@
+            + "What can I do for you?\n")
+            .concat(SEP_LINE);
+    // Level-6 -> A-Collections: Task List
+    private final List<Task> taskList = new ArrayList<>();
```
Perhaps name this TASK_LIST, since it has been declared as a constant.

```suggestion
    private final List<Task> TASK_LIST = new ArrayList<>();
```

ananya-ravi-3021 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,239 @@
+    }
+
+    // decodes input from user and passes that argument to a response builder
+    public int decoder(String userInput) {
```
Should this be changed to decodeInput to be more detailed? 
```suggestion
    public int decodeInput(String userInput) {
```

ananya-ravi-3021 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,239 @@
+            case 4:
+            case 5:
+                String desc = "";
+                String var;
```
Perhaps a more specific variable name here as well, so that it is easier for readers to understand its purpose.

ananya-ravi-3021 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,121 @@
+// Level 7: Created Storage class to maintain storage state
+public class Storage {
+
+    private final File storageFile;
```
I noticed a similar issue here as in your Duke class - perhaps name your constants using the constant naming convention?

Banana3021 reviewed (commented):
Overall, your code is very well written and your class structure/hierarchy has been well organised. Do take note of naming conventions, especially when naming constants and enums.

Otherwise, excellent work!

Dr-Octavius committed:
Use more OOP

Dr-Octavius committed:
Organize into Packages

Dr-Octavius committed:
Add A-JUnit Tests

Dr-Octavius committed:
JavaDoc

Dr-Octavius committed:
Find

Dr-Octavius committed:
Resolve Merge Conflicts

Dr-Octavius committed:
Merge branch 'add-gradle-support' of https://github.com/Dr-Octavius/ip

Dr-Octavius committed:
Use Checkstyle

Dr-Octavius committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java

Dr-Octavius committed:
Merge branch-Level-9 and branch-A-JavaDoc

Dr-Octavius committed:
Implement GUI

Dr-Octavius committed:
GUI

Dr-Octavius committed:
Merge branch-Level-10

Dr-Octavius committed:
Make Gui

Dr-Octavius committed:
GUI Changes: Make GUI refelct pictures as shown on tuytorial

Dr-Octavius committed:
GUI Changes: Make GUI refelct pictures as shown on tutorial

Dr-Octavius committed:
Task List Constructor: Able to initiate Task List from existing Storage File

Task List Constructor can only initiate new Task List in each new initialisation of a Bot

Task List Constructor: Able to initiate Task List from existing Storage File

Task List Constructor can only initiate new Task List in each new initialisation of a Bot

Dr-Octavius committed:
Revert "Task List Constructor: Able to initiate Task List from existing Storage File"

This reverts commit 07db8706f40cafdaecc2f69670233d6749f657cf.

Dr-Octavius committed:
Task List Constructor: Able to initiate Task List from existing Storage File

Task List Constructor can only initiate new Task List in each new initialisation of a Bot

Task List Constructor can now accept a .txt file as input to read in historical Task values during task list initialisation

Task List constructor that can read tasks from an existing storage .txt file allows for memory persistence across different instantiations of the Bot

Dr-Octavius committed:
Add Streams: Convert message that comes as a string form into an input stream-mMessage Strings have to be processed into arrays before it can be understood by the Bot logic

This process is faciliated by the converting the message string into an input stream and fed through a Scanner object to examine tokens

This is done so as to reduce the need for Strings to be converted into byte arrays and potentially break the existing code base which works on output processed by scanner class, which accepts stream as input

Dr-Octavius committed:
Refactor All Files to be compliant with checksum checks

Dr-Octavius committed:
Updated JavaDocs for Parser Class

Dr-Octavius committed:
Refactor All Files to be compliant with checksum checks

Dr-Octavius committed:
Add Assertions for condition checking

Dr-Octavius committed:
Implemented Sort Functionality

Dr-Octavius committed:
Set up product website

Dr-Octavius committed:
Add Second Unit Test

Dr-Octavius committed:
Delete run.sh from branch tracking

Dr-Octavius committed:
Merge pull request #3 from Dr-Octavius/branch-Level-8

Delete run.sh from branch tracking

Dr-Octavius committed:
Set theme jekyll-theme-cayman

Dr-Octavius committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

Dr-Octavius committed:
Merge branch 'master' of https://github.com/Dr-Octavius/ip
